src/generator_wds.rs

```diff
@@ -158,21 +172,25 @@ async fn pull_tarballs(
             if samples_metadata_tx.send(current_files_for_sample).is_err() {
                 debug!("dispatch_shards (streaming): samples_metadata_tx channel closed.");
                 let _ = samples_metadata_tx.close(); // Make sure that we close on both ends
-                return Err("Channel closed".into());
```

Comment:
This was not an error per say (enough samples so we closed the channel) but looked like it in the logs